pkg/api/error.go

```diff
@@ -113,11 +113,17 @@ const (
 
 // NewCloudError returns a new CloudError
 func NewCloudError(statusCode int, code, target, message string, a ...interface{}) *CloudError {
+	var msg string
```

Comment:
I reviewed some of the thousands of errors that were getting flagged here, and they were all cases where only one argument was passed and it was expected to be interpreted verbatim.

Reply:
what did linter complain?

Reply:
@weinong 

```
printf: non-constant format string in call to (*github.com/sirupsen/logrus.Entry).Warnf (govet)
```

When the first argument to `fmt.Sprintf` is not a constant string, if that value ends up having e.g. `%` literals in it, the output of `fmt.Sprintf()` will be unknown. Please see the new commits I pushed for how i solved this problem